bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerHttpApiV213.java

```diff
@@ -33,15 +36,14 @@
      * @param password Admin password for the OpenSprinkler device.
      * @param basicUsername only needed if basic auth is required
      * @param basicPassword only needed if basic auth is required
+     * @throws CommunicationApiException
      * @throws Exception
      */
     OpenSprinklerHttpApiV213(final HttpClient httpClient, final OpenSprinklerHttpInterfaceConfig config)
-            throws GeneralApiException {
-        super(httpClient, withHashedPassword(config));
-    }
-
-    private static OpenSprinklerHttpInterfaceConfig withHashedPassword(final OpenSprinklerHttpInterfaceConfig config) {
-        config.password = Hash.getMD5Hash(config.password);
-        return config;
+            throws GeneralApiException, CommunicationApiException {
+        super(httpClient, config);
+        password = Hash.getMD5Hash(password);
+        getProgramData();
+        getStationNames();
```

Comment:
Is there another way you can suggest? I did it this way because before firmware 213 these features are not supported and they also require the password to be hashed first. In older firmwares the password was not MD5 hashed.